R/pull_data.R

```diff
@@ -181,35 +181,70 @@ get_cr_data <- function() {
+    select(-c(reported_preliminary_total_potable_water_production,
+           preliminary_percent_residential_use,
+           reported_preliminary_commercial_agricultural_water,
+           reported_preliminary_commercial_industrial_and_institutional_wa,
+           county_under_drought_declaration, hydrologic_region, county,
+           water_shortage_level_indicator, dwr_state_standard_level_corresponding_to_stage,
+           calculated_r_gpcd, calculated_total_potable_water_production_gallons_ag_excluded,
+           water_shortage_contingency_stage_invoked, calculated_total_potable_water_production_gallons_2013_ag_exclu,
+           climate_zone, calculated_commercial_agricultural_water_gallons_2013, calculated_commercial_agricultural_water_gallons,
+           `_full_text`, reference_2014_population, total_population_served, `_id`, qualification)) %>%
+    pivot_longer(cols = -c(public_water_system_id, supplier_name, water_production_units, reporting_month, final_percent_residential_use),
+                 names_to = "use_type", values_to = "volume") %>%
     transmute(report_name = "CR",
-              pwsid = pwsid,
+              pwsid = public_water_system_id,
               supplier_name = supplier_name,
-              year = year,
-              month = month,
-              category = category,
+              year = lubridate::year(reporting_month),
+              month = lubridate::month(reporting_month),
+              final_percent_residential_use = final_percent_residential_use,
+              category = case_when(use_type %in% c("reported_final_total_potable_water_production",
+                                                   "calculated_total_potable_water_production_gallons_ag_excluded") ~ "supply",
+                                   use_type == "reported_non_revenue_water" ~ "demand",
+                                   TRUE ~ "demand"),
               use_type = gsub("_", " ", tolower(use_type)),
-              volume_af = ifelse(volume == "NaN", NA, as.numeric(volume)))
-
-  return(conservation_report_data)
+              volume_af = case_when(
+                water_production_units == "MG" ~ as.numeric(volume) * scale_from_MG_to_AF,
+                water_production_units == "G" ~ as.numeric(volume) * scale_from_G_to_AF,
+                water_production_units == "CCF" ~ as.numeric(volume) * scale_from_CCF_to_AF,
+                water_production_units == "AF" ~ as.numeric(volume)
+              )) |>
+    mutate(volume_af = ifelse(is.nan(volume_af), NA, volume_af))
+
+  residential_cr <- conservation_report_data |>
+    filter(category == "demand") |>
+    group_by(report_name, pwsid, supplier_name,
+             year, month, category) |>
+    summarize(total_demand = sum(volume_af, na.rm = TRUE),
+              final_prop_residential = as.numeric(final_percent_residential_use)/100,
+              residential = total_demand * final_prop_residential) |>
```

Comment:
We need to multiply by total production instead of demand:

- This is our note `% Residential (Can calculate volume residential by `Total Potable Water Production` * `Percent Residential Use`)` from https://github.com/cawaterdata/urban-water-reporting/blob/main/scripts/reports_and_metrics_overview.Rmd
- Data dictionary - https://www.waterboards.ca.gov/water_issues/programs/conservation_portal/docs/definitions_and_data_dictionary_accessible.pdf

Reply:
updated